File: chor-extraction/src/main/java/ast/sp/nodes/Receiving.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;
import ast.sp.interfaces.Interaction;

public class Receiving implements Interaction {
    private final Behaviour continuation;
    private final String process;

    public Receiving(Behaviour continuation, String process) {
        this.continuation = continuation;
        this.process = process;
    }

    public Behaviour getContinuation() {
        return continuation;
    }

    public String getProcess() {
        return process;
    }

    public String toString() {
        return process + "?; " + continuation.toString();
    }

    @Override
    public boolean findRecProcCall(String procname) {
        return continuation.findRecProcCall(procname);
    }
}

File: chor-extraction/src/main/java/ast/sp/nodes/Receiving.kt
package ast.sp.nodes

import ast.sp.interfaces.Behaviour
import ast.sp.interfaces.Interaction

data class Receiving(val continuation: Behaviour, val process: String) : Interaction {
    override fun toString(): String {
        return process + "?; " + continuation.toString()
    }

    override fun findRecProcCall(procname: String): Boolean {
        return continuation.findRecProcCall(procname)
    }

    override fun copy(): Interaction {
        return Receiving(continuation.copy(), ""+process)
    }

    override fun equals(b: Behaviour): Boolean {
        return b is Receiving &&
                b.process == process &&
                b.continuation == continuation
    }
}
